### ISSUE TITLE
A false negative about the rule IL_INFINITE_LOOP

### ISSUE THREAD

**Issue by Belle-PL** [spotbugs/spotbugs#2331]
This rule is used to detect infinite loop. SpotBugs can recognize the issue in the following code example:
```java
public void test4(boolean b) {
  while (b = true) {  // can report a warning
    System.out.println("Wow");
  } 
}
```

But the following example cannot be detected, these two cases are equivalent.

```java
public void test4(boolean b) {
  do {
    System.out.println("Wow");
  } while (b = true); // should report a warning here
}
```

Version: LATEST

**Comment by yuliyasokh** [comment]
Hello @Belle-PL 
I was not able to reproduce this issue on version 4.7.3 
Please check if issue is still persists for you with latest version 

![loop](https://github.com/spotbugs/spotbugs/assets/2944397/49f403be-11d9-4aa2-9abe-5724ca86f420)

**Comment by JuditKnoll** [comment]
According to https://github.com/spotbugs/spotbugs/pull/2518, this issue is solved.
